server/auth.py

```
@@ -126,8 +126,13 @@ def check(conn: sqlite3.Connection, token: str | None) -> tuple[str, str] | None
     if int(expiry) < time.time():
         return None
     row = conn.execute(
-        "SELECT role FROM users WHERE username = ?", (username,)
+        "SELECT role, disabled_at FROM users WHERE username = ?", (username,)
     ).fetchone()
     if row is None or not row["role"]:
         return None                # נמחק, או שורה בלי תפקיד — אין הרשאה
+    if row["disabled_at"]:
+        # ‏#186: חסימה חלה **כאן** ולא רק בכניסה. חסימה שנבדקת בכניסה
+        # בלבד הייתה פותחת מחדש את החלון שסגר #91 — עד 12 שעות שבהן
```

Ruff (0.16.2)

[error] Contains control characters that can permit obfuscated code

(PLE2502)